File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--programmatically-store-and-retrieve-date-values-in-excel-ranges_3.vb
        Dim value As Object = NamedRange1.Value2

        If Not value Is Nothing Then
            If TypeOf value Is Double Then
                dt = DateTime.FromOADate(CType(value, Double))
            Else
                DateTime.TryParse(CType(value, String), dt)
            End If
        End If

        MessageBox.Show(dt.ToString())

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--programmatically-store-and-retrieve-date-values-in-excel-ranges_3.cs
            object value = NamedRange1.Value2;

            if (value != null)
            {
                if (value is double)
                {
                    dt = DateTime.FromOADate((double)value);
                }
                else
                {
                    DateTime.TryParse((string)value, out dt);
                }
            }
            MessageBox.Show(dt.ToString());